lib/LaTeXML/Core/Gullet.pm

```diff
@@ -270,8 +270,6 @@ sub readXToken {
         # the token leaves the pushback.
         # This is *required to be different* from the noexpand flag, as per the B Book
         @expansion = map { T_NOEXPAND1($_); } @expansion; }
-      else {
-        @expansion = map { ($$_[1] == CC_NOEXPAND1 ? $$_[2] : $_); } @expansion; }
```

Comment:
Similar question as below: Is it best to remove them as early as possible, or try to force a fail?

Also: with the modfied `DefExpanded`, I was able to prefix `$LaTeXML::KEEP_THE &&` into the test clause just ahead of this. Again, the point was to limit the creation & scope of NOEXPAND.

Reply:
Certain types of failures are valuable in debugging, and being able to get a Fatal for a double-wrap operation should be quite helpful in failing earlier than one would have otherwise. If you unwrapped them early, and then re-wrapped wrongly (due to some unforeseen logic mistake), it will be quite some pain to figure out what happened. Unwrapping only where that was *meant* in a TeX sense, and loudly erroring otherwise, makes debugging a lot more robust in my view.

Reply:
My comment assumes I will be back to debugging expl3 in texlive 2021 btw, to situate my debugging claims into some specifics. But I think this may also help in bits like tikz and xy, if they happen to rely on similar no-expansion-related infrastructure.

Reply:
> Similar question as below: Is it best to remove them as early as possible, or try to force a fail?
> 
> Also: with the modfied `DefExpanded`, I was able to prefix `$LaTeXML::KEEP_THE &&` into the test clause just ahead of this. Again, the point was to limit the creation & scope of NOEXPAND.

I appreciate the sentiment, but that gets further from what TeX does. The core `expand` procedure (366. and 367. in the B Book) will always perform a special `ins_the_toks` call to insert/smuggle in tokens returned from `\the` and its friends. Those tokens were respectively stored in a temp variable `the_toks`. We're emulating the temp variable by setting the new special catcode and unshifting in the gullet's `{pushback}`, which is accurate enough (I think). But importantly, according to 367., this is done for all cases where `\the` is called, not just for the definitional ones (where the catcodes are also "kept" outside of gullet).